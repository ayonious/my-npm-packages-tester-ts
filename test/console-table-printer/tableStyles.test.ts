import { expect } from '@jest/globals';
import { Table } from 'console-table-printer';

describe('Table with different styles', () => {
  it('should render table with title', () => {
    const table = new Table({
      title: 'Table with title',
      columns: [
        { name: 'id', alignment: 'left' },
        { name: 'name', alignment: 'center' },
        { name: 'score', alignment: 'right' },
      ],
    });

    table.addRows([
      { id: 1, name: 'Alice', score: 95 },
      { id: 2, name: 'Bob', score: 88 },
      { id: 3, name: 'Charlie', score: 92 },
    ]);

    expect(table.render()).toMatchSnapshot();
  });

  it('should render table with custom colors', () => {
    const table = new Table({
      title: 'Table with custom colors',
      columns: [
        { name: 'id', alignment: 'left', color: 'blue' },
        { name: 'name', alignment: 'center', color: 'green' },
        { name: 'score', alignment: 'right', color: 'yellow' },
      ],
    });

    table.addRows([
      { id: 1, name: 'Alice', score: 95 },
      { id: 2, name: 'Bob', score: 88 },
      { id: 3, name: 'Charlie', score: 92 },
    ]);

    expect(table.render()).toMatchSnapshot();
  });

  it('should render table with custom column titles', () => {
    const table = new Table({
      title: 'Table with custom column titles',
      columns: [
        { name: 'id', title: 'ID', alignment: 'left' },
        { name: 'name', title: 'Full Name', alignment: 'center' },
        { name: 'score', title: 'Test Score', alignment: 'right' },
      ],
    });

    table.addRows([
      { id: 1, name: 'Alice', score: 95 },
      { id: 2, name: 'Bob', score: 88 },
      { id: 3, name: 'Charlie', score: 92 },
    ]);

    expect(table.render()).toMatchSnapshot();
  });

  it('should render table with custom color map', () => {
    const table = new Table({
      title: 'Table with custom color map',
      columns: [
        { name: 'id', alignment: 'left' },
        { name: 'name', alignment: 'center' },
        { name: 'score', alignment: 'right' },
      ],
      colorMap: {
        custom_green_bold: '\x1b[1m\x1b[32m',
      },
    });

    table.addRows([{ id: 1, name: 'Alice', score: 95 }], {
      color: 'custom_green_bold',
    });

    table.addRows(
      [
        { id: 2, name: 'Bob', score: 88 },
        { id: 3, name: 'Charlie', score: 92 },
      ],
      { color: 'yellow' }
    );

    expect(table.render()).toMatchSnapshot();
  });
});
