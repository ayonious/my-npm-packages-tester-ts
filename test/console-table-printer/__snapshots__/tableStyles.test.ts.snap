// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Table with different styles should render table with custom color map 1`] = `
"[01mTable with custom color 
          map           [0m
┌────┬─────────┬───────┐
│[37m [0m[01mid[0m │[37m [0m[01m name  [0m │[37m [0m[01mscore[0m │
├────┼─────────┼───────┤
│[37m [0m[1m[32m1 [0m │[37m [0m[1m[32m Alice [0m │[37m [0m[1m[32m   95[0m │
│[37m [0m[33m2 [0m │[37m [0m[33m  Bob  [0m │[37m [0m[33m   88[0m │
│[37m [0m[33m3 [0m │[37m [0m[33mCharlie[0m │[37m [0m[33m   92[0m │
└────┴─────────┴───────┘"
`;

exports[`Table with different styles should render table with custom colors 1`] = `
"[01mTable with custom colors[0m
┌────┬─────────┬───────┐
│[37m [0m[01mid[0m │[37m [0m[01m name  [0m │[37m [0m[01mscore[0m │
├────┼─────────┼───────┤
│[37m [0m[34m1 [0m │[37m [0m[32m Alice [0m │[37m [0m[33m   95[0m │
│[37m [0m[34m2 [0m │[37m [0m[32m  Bob  [0m │[37m [0m[33m   88[0m │
│[37m [0m[34m3 [0m │[37m [0m[32mCharlie[0m │[37m [0m[33m   92[0m │
└────┴─────────┴───────┘"
`;

exports[`Table with different styles should render table with custom column titles 1`] = `
"[01mTable with custom column titles[0m
┌────┬───────────┬────────────┐
│[37m [0m[01mID[0m │[37m [0m[01mFull Name[0m │[37m [0m[01mTest Score[0m │
├────┼───────────┼────────────┤
│[37m [0m[37m1 [0m │[37m [0m[37m  Alice  [0m │[37m [0m[37m        95[0m │
│[37m [0m[37m2 [0m │[37m [0m[37m   Bob   [0m │[37m [0m[37m        88[0m │
│[37m [0m[37m3 [0m │[37m [0m[37m Charlie [0m │[37m [0m[37m        92[0m │
└────┴───────────┴────────────┘"
`;

exports[`Table with different styles should render table with title 1`] = `
"[01m    Table with title    [0m
┌────┬─────────┬───────┐
│[37m [0m[01mid[0m │[37m [0m[01m name  [0m │[37m [0m[01mscore[0m │
├────┼─────────┼───────┤
│[37m [0m[37m1 [0m │[37m [0m[37m Alice [0m │[37m [0m[37m   95[0m │
│[37m [0m[37m2 [0m │[37m [0m[37m  Bob  [0m │[37m [0m[37m   88[0m │
│[37m [0m[37m3 [0m │[37m [0m[37mCharlie[0m │[37m [0m[37m   92[0m │
└────┴─────────┴───────┘"
`;
